samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -19,32 +19,57 @@
 package org.apache.samza.zk;
 
 import com.google.common.base.Strings;
+import java.util.Map;
 import org.I0Itec.zkclient.ZkClient;
 import org.apache.samza.SamzaException;
 import org.apache.samza.config.Config;
+import org.apache.samza.config.MetricsConfig;
 import org.apache.samza.config.ZkConfig;
 import org.apache.samza.coordinator.CoordinationServiceFactory;
 import org.apache.samza.coordinator.CoordinationUtils;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.util.MetricsReporterLoader;
 import org.apache.zookeeper.client.ConnectStringParser;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-  private final static Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
+
+  private static final String SOURCE_PREFIX = "ZkJobCoordinationService";
 
   public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
 
     ZkClient zkClient =
         createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
 
-    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
+    ZkJobCoordinatorMetrics zkJobCoordinatorMetrics = getJobCoordinatorMetrics(config, participantId);
+
+    ZkUtils zkUtils = new ZkUtils(new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs(), zkJobCoordinatorMetrics);
 
     return new ZkCoordinationUtils(participantId, zkConfig, zkUtils);
   }
 
   /**
+   * Builds {@link ZkJobCoordinatorMetrics} and registers it with MetricsReporters.
+   * @param config required to build ZkJobCoordinatorMetrics.
+   * @param participantId a unique id that identifies the participant.
+   * @return the built ZkJobCoordinatorMetrics.
+   */
+  private ZkJobCoordinatorMetrics getJobCoordinatorMetrics(Config config, String participantId) {
```

Comment:
Agreed. Update the patch based upon offline discussions.

Passed in ZkJobCoordinatorMetrics with NoopsMetricsRegistry, instead of null. Let's create a JIRA to revisit this in the future.